### PR TITLE
Add CheckWitness to func invokefunction

### LINKED ISSUE #212: CheckWitness on invokefunction
port to Neo3 https://github.com/neo-project/neo/pull/1418

### PR DESCRIPTION
**Pull request by Qiao-Jin**
Close #212

### CODE REVIEW

File: src/RpcServer/RpcServer.SmartContract.cs [hunk 1]
```diff
@@ -76,12 +76,18 @@ private JObject InvokeFunction(JArray _params)
             UInt160 script_hash = UInt160.Parse(_params[0].AsString());
             string operation = _params[1].AsString();
             ContractParameter[] args = _params.Count >= 3 ? ((JArray)_params[2]).Select(p => ContractParameter.FromJson(p)).ToArray() : new ContractParameter[0];
+            CheckWitnessHashes checkWitnessHashes = null;
+            if (_params.Count > 3)
+            {
+                UInt160[] scriptHashesForVerifying = _params.Skip(3).Select(u => UInt160.Parse(u.AsString())).ToArray();
+                checkWitnessHashes = new CheckWitnessHashes(scriptHashesForVerifying);
+            }
```
Comment (erikzhang) [inline src/RpcServer/RpcServer.SmartContract.cs:84]: The new parameter can be an array.

Comment (Qiao-Jin) [reply src/RpcServer/RpcServer.SmartContract.cs]: Have modified correspondingly:)

Comment (shargon) [reply src/RpcServer/RpcServer.SmartContract.cs]: I prefer accept only array, it's easier to understand.

Comment (erikzhang) [reply src/RpcServer/RpcServer.SmartContract.cs]: Array only please. Otherwise it is impossible to add new parameters in the future.

Comment (Qiao-Jin) [reply src/RpcServer/RpcServer.SmartContract.cs]: Now only JArray format is reserved.

**Comment by ProDog** [comment]
I have tested it passed. @erikzhang plz merge it.

image

**Pull request merged**